### PR TITLE
SIM: MC truth interfaces clean-up

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
An iteration to clean-up interfaces to MC truth is provided. Extra data fields are added to MC truth but algorithm itself is not changed. Full regression is expected.

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Notification/interface/TrackWithHistory.h [hunk 1]
```diff
@@ -57,11 +60,20 @@ class TrackWithHistory {
   const math::XYZTLorentzVectorF &getMomentumAtBoundary() const { return momentumAtBoundary_; }
   int getIDAtBoundary() const { return idAtBoundary_; }
 
+  // tracker surface
+  const math::XYZVectorD &trackerSurfacePosition() const { return tkSurfacePosition_; }
```
Comment (fabiocos) [inline SimG4Core/Notification/interface/TrackWithHistory.h:64]: @civanch where do you intend to use it? I fail to see its use in this PR

Comment (civanch) [reply SimG4Core/Notification/interface/TrackWithHistory.h:64]: @fabiocos , this surface data are a part of SimTrack - persistent MC truth. Until now these was not filled at all and unused. In the current PR I have added the same variables to TrackWithHistory in order to allow Phase2 detectors to use this for a good purpose. The name of course misleading, may be we would change SimTrack somehow adding information needed both for MTD and HGCal.

Comment (civanch) [reply SimG4Core/Notification/interface/TrackWithHistory.h:64]: In the next iteration I want to drop TmpSimTrack and use TrackWithHistory to fill SimTrack.

**Pull request merged**